File: leetcode/array/2391-Minimum-Amount-of-Time-to-Collect-Garbage.cpp
class Solution {
public:
    int garbageCollection(vector<string>& garbage, vector<int>& travel) {
        int pickup = 0;
        int lastm = 0;
        int lastp = 0;
        int lastg = 0;
        for (int i = 0; i < garbage.size(); i++) {
            for (char c : garbage[i]) {
                pickup++;
                if (c == 'M')
                    lastm = i;
                if (c == 'P')
                    lastp = i;
                if (c == 'G')
                    lastg = i;
            }
        }
        cout << lastm;
        cout << lastg;
        cout << lastp;
        int gtime = 0;
        int ptime = 0;
        int mtime = 0;
        int time = 0;
        for (int i = 0; i < travel.size(); i++) {
            time += travel[i];
            if (i < lastm)
                mtime += travel[i];
            if (i < lastp)
                ptime += travel[i];
            if (i < lastg)
                gtime += travel[i];
        }

        return gtime + ptime + mtime + pickup;
    }
};

File: leetcode/array/2391-Minimum-Amount-of-Time-to-Collect-Garbage.py
class Solution:
    def __init__(self):
        self.hasG = False
        self.hasP = False
        self.hasM = False

    def garbageCollection(self, garbage, travel):
        n = len(garbage)
        res = 0

        for i in range(n - 1):
            res += 3 * travel[i]

        for s in garbage:
            res += len(s)

        for i in range(n - 1, 0, -1):
            if "G" not in garbage[i]:
                res -= travel[i - 1]
            else:
                break

        for i in range(n - 1, 0, -1):
            if "P" not in garbage[i]:
                res -= travel[i - 1]
            else:
                break

        for i in range(n - 1, 0, -1):
            if "M" not in garbage[i]:
                res -= travel[i - 1]
            else:
                break

        return res